src/network/MissionControlTasks.cpp

```diff
@@ -141,6 +142,23 @@ void CameraStreamTask::closeStream(const CameraID& cam) {
 	_camera_encoders.erase(cam);
 }
 
+void CameraStreamTask::sendCurrentFrame(const robot::types::CameraID& cam) {
+	std::string b64_data = "";
+	{
+		std::lock_guard lg(_mutex);
+		if (_open_streams.find(cam) != _open_streams.end()) {
+			auto camDP = robot::readCamera(cam);
+			if (camDP) {
+				auto data = camDP.getData();
+				cv::Mat frame = data.first;
+				b64_data = base64::encodeMat(frame, ".jpg");
+			}
```

Comment:
If the `camDP` datapoint is empty, this will respond with an empty string. Instead, we should probably not send a response at all.